fuel-tx/src/transaction/types/input.rs

```diff
@@ -55,20 +56,65 @@ pub trait AsField<Type>: AsFieldFmt {
     fn as_mut_field(&mut self) -> Option<&mut Type>;
 }
 
-impl<Type> AsField<Type> for () {
+/// The empty field used by sub-types of the specification.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Empty<Type>(::core::marker::PhantomData<Type>);
+
+impl<Type> Empty<Type> {
+    /// Creates `Self`.
+    pub const fn new() -> Self {
+        Self(::core::marker::PhantomData {})
+    }
+}
+
+impl<Type> Default for Empty<Type> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<Type: Serialize + Default> Serialize for Empty<Type> {
     #[inline(always)]
-    fn as_field(&self) -> Option<&Type> {
-        None
+    fn size_static(&self) -> usize {
+        Type::default().size_static()
     }
 
-    fn as_mut_field(&mut self) -> Option<&mut Type> {
-        None
+    #[inline(always)]
+    fn size_dynamic(&self) -> usize {
+        0
+    }
+
+    #[inline(always)]
+    fn encode_static<O: Output + ?Sized>(&self, buffer: &mut O) -> Result<(), Error> {
+        Type::default().encode_static(buffer)
```

Comment:
If this is empty, then it probably shouldn't even write anything to the buffer at all right?

Reply:
In the case of the `Vec`, it writes 8 bytes - the length of the vector with the value `0`. The compiler will optimize it into writing zero into the array.

The `Empty` field doesn't occupy memory in runtime, but during serialization and deserialization it should participate and we need to include the static part.